ui/utils.py

```diff
@@ -170,3 +179,148 @@ def write_x509_files():
+    END_DATE = '9999-01-01'
+    query = {
+        'reportRequests': [
+            {
+                'viewId': settings.GA_VIEW_ID,
+                'dateRanges': [{'startDate': START_DATE, 'endDate': END_DATE}],
+                'metrics': [{'expression': 'ga:totalEvents'}],
+                'dimensions': [
+                    # [adorsk, 2018-03]
+                    # Achtung! a high degree of implicit coupling to
+                    # dimension names that have been manually set in GA.
+                    # Hard-coding for now.
+                    {'name': 'ga:eventAction'},
+                    {'name': 'ga:' + settings.GA_DIMENSION_CAMERA}
+                ],
+                'dimensionFilterClauses': [
+                    {
+                        'filters': [
+                            {
+                                'dimensionName': 'ga:eventLabel',
+                                'operator': 'EXACT',
+                                # 2018-03, dorsk
+                                # Achtung! We do video_key.capitalize()
+                                # because GA has capitalized event data,
+                                # due to an unexpected side-effect of the
+                                # react-ga library.
+                                # See: https://github.com/mitodl/odl-video-service/pull/472
+                                'expressions': [video_key.capitalize()]
+                            },
+                        ],
+                    }
+                ],
+            },
+        ],
+    }
+    return query
+
+
+def parse_google_analytics_response(ga_response):
+    """Parse a Google Analytics response.
+
+    Returns:
+        data: a dict of parsed response data, in this shape:
+            {
+                'times': [0, ..., N], # list of all times
+                'channels': ['camera0', ..., 'cameraN'], # list of all channels
+                'views_at_times': {
+                    0: {
+                        'camera0': 23,
+                        'camera1': 3,
+                        ...
+                        <cameraN>: <X>
+                    },
+                    ...
+                    <timeN>: <views_per_channel_at_timeN>
+                },
+            }
+    """
+    times = set()
+    channels = set()
+    views_at_times = {}
+    rows = ga_response['reports'][0]['data'].get('rows', [])
+    for row in rows:
+        m = re.match(r'T(\d{4})', row['dimensions'][0])
+        if not m:
+            continue
+        time_ = int(m.group(1))
+        channel = row['dimensions'][1]
+        viewers = int(row['metrics'][0]['values'][0])
+        views_at_times.setdefault(time_, {}).update({channel: viewers})
+        times.add(time_)
+        channels.add(channel)
+    return {
+        'times': sorted(list(times)),
+        'channels': sorted(list(channels)),
+        'views_at_times': views_at_times,
+    }
+
+
+def generate_mock_video_analytics_data(n=24, seed=42):
```

Comment:
Could this go in one of the test files instead (like `views_test.py`)?

Reply:
I think this depends on whether we decide to remove the 'mock' parameter (per above comment).

If we decide to remove it, I agree, we should move this out.

If we decide to leave it, I think we should keep this here, because it's part of a 'real' view.